src/R/Core/Impl/AST/Extensions/AstRootExtensions.cs

```diff
@@ -38,5 +39,18 @@ public static class AstRootExtensions {
             }
             return null;
         }
+
+        public static bool IsCompleteExpression(this AstRoot expressionAst) {
+            foreach (var error in expressionAst.Errors) {
+                if (error.ErrorType == ParseErrorType.CloseCurlyBraceExpected ||
+                    error.ErrorType == ParseErrorType.CloseBraceExpected ||
+                    error.ErrorType == ParseErrorType.CloseSquareBracketExpected ||
+                    error.ErrorType == ParseErrorType.FunctionBodyExpected ||
```

Comment:
This is code we used to use in REPL to show continuation. But no, it does not currently detect missing body on scope-less 'if' or similar.